pkg/oc/cli/admin/mustgather/mustgather.go

```diff
@@ -0,0 +1,330 @@
+		Use:     "must-gather",
+		Short:   "Launch a new instance of a pod for gathering debug information",
+		Long:    mustGatherLong,
+		Example: mustGatherExample,
+		Hidden:  true,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run(rsyncCommand))
+		},
+	}
+
+	cmd.Flags().StringVar(&o.NodeName, "node-name", "", "Set a specific node to use - by default a random master will be used")
+	cmd.Flags().StringVar(&o.Image, "image", "", "Set a specific image to use, by default the OpenShift's must-gather image will be used.")
+	cmd.Flags().StringVar(&o.DestDir, "dest-dir", "", "Set a specific directory on the local machine to write gathered data to.")
+	cmd.Flags().BoolVar(&o.Keep, "keep", false, "Do not delete temporary resources when command completes.")
+	_ = cmd.Flags().MarkHidden("keep")
+
+	return cmd
+}
+
+func (o *Options) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	if i := cmd.ArgsLenAtDash(); i != -1 && i < len(args) {
+		o.Command = args[i:]
+	} else {
+		o.Command = args
+	}
+
+	var err error
+
+	if o.Config, err = f.ToRESTConfig(); err != nil {
+		return err
+	}
+
+	if o.Client, err = kubernetes.NewForConfig(o.Config); err != nil {
+		return err
+	}
+
+	if len(o.DestDir) == 0 {
+		o.DestDir = fmt.Sprintf("must-gather.local.%06d", rand.Int63())
+	}
+
+	if len(o.Image) == 0 {
+		imageClient, err := v1.NewForConfig(o.Config)
+		if err != nil {
+			return err
+		}
+		imageStream, err := imageClient.ImageStreams("openshift").Get("must-gather", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		var ok bool
+		if o.Image, ok = util.ResolveLatestTaggedImage(imageStream, "latest"); !ok {
+			return fmt.Errorf("unable to to resolve must-gather image")
+		}
+	}
+
+	o.RsyncRshCmd = rsync.DefaultRsyncRemoteShellToUse(cmd.Parent())
+
+	return nil
+}
+
+type Options struct {
+	genericclioptions.IOStreams
+
+	Config *rest.Config
+	Client kubernetes.Interface
+
+	NodeName string
+	DestDir  string
+	Image    string
+	Command  []string
+	Keep     bool
+
+	RsyncRshCmd string
+}
+
+// Run creates and runs a must-gather pod.d
+func (o *Options) Run(rsyncCmd *cobra.Command) error {
+	if len(o.Config.BearerToken) != 0 {
+		return fmt.Errorf("cannot run must-gather with a token based kubeconfig")
+	}
+	if len(o.Image) == 0 {
+		return fmt.Errorf("missing an image")
+	}
+
+	var err error
+
+	// create namespace
+	ns, err := o.Client.CoreV1().Namespaces().Create(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "openshift-must-gather-",
+			Labels: map[string]string{
+				"openshift.io/run-level": "0",
+			},
+			Annotations: map[string]string{
+				"oc.openshift.io/command": "oc adm must-gather",
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if !o.Keep {
+		defer func() {
+			err = o.Client.CoreV1().Namespaces().Delete(ns.Name, nil)
+		}()
+	} else {
+		fmt.Fprintf(o.Out, "namespace/%s created.\n", ns.Name)
```

Comment:
I'd prefer we use `NamePrinter` for that, create `printers.NewPrinter` and use that here and in the ClusterRoleBinding printing. This will give us consistent output, especially when we'll go crazy and decide to change the format, again :wink: